gnovm/pkg/gnolang/store.go

```diff
@@ -488,6 +489,39 @@ func (ds *defaultStore) SetObject(oo Object) {
 		panic("should not happen")
 	}
 	oo.SetHash(ValueHash{hash})
+	// make store op log entry
+	if ds.opslog != nil {
+		obj := o2.(Object)
+		if oo.GetIsNewReal() {
+			fmt.Fprintf(ds.opslog, "c[%v]=%s\n",
+				obj.GetObjectID(),
+				prettyJSON(amino.MustMarshalJSON(obj)))
+		} else {
+			old := ds.loadForLog(oid)
+			old.SetHash(ValueHash{})
```

Comment:
Think It might be costly to load and marshal the old object. Since the opslog is mainly for debugging(?), should we introduce a flag to control what get recorded and printed? thinking...

Reply:
The `// Realm:` directive is used in a few dozen tests, and loading/marshaling of the old object isn't that expensive. My empirical feeling is that it doesn't make the tests significantly slow, though I'm open to alternatives

Reply:
My point is that we could do something like this:
```go
if testMode {
    // Record opslog...
}
```
This way, the opslog functionality would only be active during tests. Do you think it’s worth doing?

Reply:
The opslog is already enabled only during tests :)

This is not strictly enforced, but see the references for `SetLogStoreOps`: it is only activated now specifically to when we run filetests with `// Realm`

Reply:
yes right, thanks :D